analyzer/src/main/scala/com/avsystem/commons/analyzer/ThrownExceptionNotInFunction.scala

```diff
@@ -0,0 +1,18 @@
+package com.avsystem.commons
+package analyzer
+
+import scala.tools.nsc.Global
+
+final class ThrownExceptionNotInFunction(g: Global) extends AnalyzerRule(g, "thrownExceptionNotInFunction") {
```

Comment:
The name `ThrownExceptionNotInFunction` suggests it checks for exceptions thrown outside functions, but it’s more about misusing thrown exceptions as function arguments.

Good rule names tend to:
- Describe the smell, not the fix, so users scan a list of rules to see what they catch.
- Include the syntactic location (“…Argument”, “…Literal”, “…Parameter”). This avoids ambiguity with similar rules.
- Stay concise
- Use positive form if possible - easier to skim than “Not”/“Missing”

Therefore, I'd propose sth like `ThrowAsFunctionArgument`. It's precise (the problem is exactly a throw used as an argument where a function is expected) and reads naturally.